bin/asl-setup-dkms-mok

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/bash
+#
+
+if [ "${USER}" != "root" ]; then
+	echo "Must be run as root"
+	exit 1
+fi
+
+if [ ! -x /usr/bin/mokutil ]; then
+	echo "Missing /usr/bin/mokutil"
+	exit 1
+fi
+
+SBSTATE=$(mokutil --sb-state | awk '{print $2}')
+if [ "${SBSTATE}" != "enabled" ]; then
+	echo "This system is not running SecureBoot"
+	exit 0
+fi
+
+echo "!!!"
+echo "!!! WARNING - RUNNING THIS SCRIPT WILL REGENERATE THE"
+echo "!!! DEFAULT MACHINE OWNER KEY (MOK) FOR SECURE BOOT. IN GENERAL YOU"
+echo "!!! ARE RUNNING THIS SCRIPT BECAUSE YOU NEED TO AND YOU KNOW WHY"
+echo "!!! YOU ARE DOINGING IT. HOWEVER BE WARNED!"
+echo "!!!"
+read -p "ENTER \"galaxy\" TO CONTINUE>> " proceed
+
+if [ "${proceed}" != "galaxy" ]; then
+	exit 1
+fi
+
+echo ""
+echo "Generating an MOK requires a name for the certificate. This can be"
+echo "anything, however your name or your callsign is recommended."
+echo ""
+read -p "Name or callsign>> " subj
+
+if [ -z "${subj}" ]; then
+	echo "MOK subject cannot be blank"
+	exit 1
+fi
+
+echo ""
+echo "Generating the MOK..."
+test ! -d /var/lib/shim-signed/mok && mkdir -p /var/lib/shim-signed/mok
```

Comment:
`mkdir -p` will not fail if the directory already exists, so why use `test ! -d`?

Reply:
Good point